### PR TITLE
[NDK] Rename Reserved to MaximumProcessors in SYSTEM_PROCESSOR_INFORMATION

### PR DESCRIPTION
**Pull request by alexhenrie**
This field is the current number of processors in the system plus the number of processors that could be added without rebooting.

https://www.geoffchappell.com/studies/windows/km/ntoskrnl/api/ex/sysinfo/processor.htm

### CODE REVIEW

File: sdk/include/ndk/extypes.h [hunk 1]
```diff
@@ -756,7 +756,7 @@ typedef struct _SYSTEM_PROCESSOR_INFORMATION
     USHORT ProcessorArchitecture;
     USHORT ProcessorLevel;
     USHORT ProcessorRevision;
-    USHORT Reserved;
+    USHORT MaximumProcessors;
```
Comment (HBelusca) [inline sdk/include/ndk/extypes.h:759]:
```suggestion
#if (NTDDI_VERSION < NTDDI_WIN8)
    USHORT Reserved;
#else
    USHORT MaximumProcessors;
#endif
```

File: ntoskrnl/ex/sysinfo.c [hunk 2]
```diff
@@ -657,7 +657,7 @@ QSI_DEF(SystemProcessorInformation)
     Spi->ProcessorArchitecture = KeProcessorArchitecture;
     Spi->ProcessorLevel = KeProcessorLevel;
     Spi->ProcessorRevision = KeProcessorRevision;
-    Spi->Reserved = 0;
+    Spi->MaximumProcessors = 0;
```
Comment (HBelusca) [inline ntoskrnl/ex/sysinfo.c:660]:
```suggestion
#if (NTDDI_VERSION < NTDDI_WIN8)
    Spi->Reserved = 0;
#else
    Spi->MaximumProcessors = 0;
#endif
```

**Review by HBelusca — COMMENTED** [review]
image

**Comment by HBelusca** [comment]
Thanks for your PR! I surrounded the member by Win8+ guards since this member obtained its actual purpose there (and we usually try to keep track of these sort of things in the NDK).

**Comment by alexhenrie** [comment]
Thank you!

**Pull request merged**